noxfile.py

```diff
@@ -0,0 +1,62 @@
+import nox
+
+nox.options.reuse_existing_virtualenvs = True
+nox.options.stop_on_first_error = True
+
+deps = [
+  "setuptools-rust",
+  "fixtures",
+  "testtools>=2.5.0",
+  "networkx>=2.5",
+  "stestr",
+]
+
+extras = [
+  "mpl",
+  "graphviz",
+]
+
+@nox.session(python=["3"])
```

Comment:
This is for @mtreinish: we could add all Python versions here, but I feel the way our CI is structured we prefer to spawn different jobs rather than create multiple Python environments in the same job

Reply:
For me a common use case is I do things like `tox -epy38 && tox -epy311` to test min and max versions. As long as this is supported in some way it's fine. I've never been a fan of the tox envlist approach where you call `tox` bare and it runs the list of jobs.

Reply:
I added support for that with `nox --python 3.11 -e test_with_version`. That way `nox -e test` triggers the version specified in CI and the `test_with_version` launches a specific version (or all of them)